designs/resource-properties.md

```diff
@@ -493,3 +386,111 @@ one resource lifecycle operation’s input or output.
 * @cfnCollection trait and append versus replace semantics
 * Full design and implementation of tag traits
 
+## Alternative approaches
+
+### (`@notProperty` alternative) Add `ignoredProperties` to resource
+
+Rather than define the `@notProperty` trait on members, it could be defined on
+the resource directly:
+
+```
+resource Config {
+    identifiers: {
+        configId: ConfigId
+    }
+    properties: {
+        configData: ConfigData
+        name: ConfigName
+        tags: TagList
+        configArn: ConfigArg
+        configType: ConfigType
+    }
+    ignoredProperties: {
+        token: String
+    }
+    create: CreateConfig
+}
+```
+
+This could be made more succinct with mixins:
+
+```
+@mixin
+resource BaseResourceMixin {
+    ignoredProperties: {
+        token: String
+    }
+}
+
+resource Config with [BaseResourceMixin] {
+    identifiers: {
+        configId: ConfigId
+    }
+    properties: {
+        configData: ConfigData
+        name: ConfigName
+        tags: TagList
+        configArn: ConfigArg
+        configType: ConfigType
+    }
+    create: CreateConfig
+}
+```
+
+Pros:
+
+* Remove the need to add property traits to members spread across the resource.
+
+Cons:
+* Marks
+
+* The resource shape becomes very verbose.
+* Mixins can also be used with inputs and outputs to mark specific members as
```

Comment:
This isn't really a con though. I think a different con is that it's easier to put this trait directly on deviating members, and it removes the need to do more intricate model transformation if a resource operation is filtered out of a projection (e.g., you added an exclusion for a property for a specific operation, but that operation is in private beta, so now you have to modify the exclusion list and remove the operation, which is error prone).